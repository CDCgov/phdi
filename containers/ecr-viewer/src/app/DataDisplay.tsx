import React, { ReactNode, useEffect, useState } from "react";
import { Button } from "@trussworks/react-uswds";
import classNames from "classnames";

import { ToolTipElement } from "@/app/ToolTipElement";

export interface DisplayDataProps {
  title?: string;
  className?: string;
  toolTip?: string;
  value?: string | React.JSX.Element | React.JSX.Element[] | React.ReactNode;
  dividerLine?: boolean;
}

/**
 * Functional component for displaying data.
 * @param props - Props for the component.
 * @param props.item - The display data item to be rendered.
 * @param [props.className] - Additional class name for styling purposes.
 * @returns - A React element representing the display of data.
 */
export const DataDisplay: React.FC<{
  item: DisplayDataProps;
  className?: string;
}> = ({
  item,
  className,
}: {
  item: DisplayDataProps;
  className?: string;
}): React.JSX.Element => {
  item.dividerLine =
    item.dividerLine == null || item.dividerLine == undefined
      ? true
      : item.dividerLine;
  return (
    <div>
      <div className="grid-row">
        <div className="data-title">
          <ToolTipElement content={item.title} toolTip={item.toolTip} />
        </div>
        <div
          className={classNames(
            "grid-col-auto maxw7 text-pre-line",
            className,
            item.className ? item.className : "",
          )}
        >
          <FieldValue>{item.value}</FieldValue>
        </div>
      </div>
      {item.dividerLine ? <div className={"section__line_gray"} /> : ""}
    </div>
  );
};
/**
 * Functional component for displaying a value. If the value has a length greater than 500 characters, it will be split after 300 characters with a view more button to view the entire value.
 * @param value - props for the component
 * @param value.children - the value to be displayed in the value
 * @returns - A React element representing the display of the value
 */
const FieldValue: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  const maxLength = 500;
  const cutLength = 300;
  const [hidden, setHidden] = useState(true);
  const [fieldValue, setFieldValue] = useState(children);
  const valueLength = getReactNodeLength(children);
  const cutField = trimField(children, cutLength, setHidden).value;
  useEffect(() => {
    if (valueLength > maxLength) {
      if (hidden) {
        setFieldValue(cutField);
      } else {
        setFieldValue(
          <>
            {children}&nbsp;
            <Button
              type={"button"}
              unstyled={true}
              onClick={() => setHidden(true)}
            >
              View less
            </Button>
          </>,
        );
      }
    }
  }, [hidden]);

  return fieldValue;
};
/**
 * Recursively determine the character length of a ReactNode
 * @param value - react node to be measured
 * @returns - the number of characters in the ReactNode
 */
const getReactNodeLength = (value: React.ReactNode): number => {
  if (typeof value === "string") {
    return value.length;
  } else if (Array.isArray(value)) {
    let count = 0;
    value.forEach((val) => (count += getReactNodeLength(val)));
    return count;
  } else if (React.isValidElement(value) && value.props.children) {
    return getReactNodeLength(value.props.children);
  }
  return 0;
};
/**
 * Create an element with `remainingLength` length followed by a view more button
 * @param value - the value that will be cut
 * @param remainingLength - the length of how long the returned element will be
 * @param setHidden - a function used to signify that the view more button has been clicked.
 * @returns - an object with the shortened value and the length left over.
 */
const trimField = (
  value: React.ReactNode,
  remainingLength: number,
  setHidden: (val: boolean) => void,
): { value: React.ReactNode; remainingLength: number } => {
  if (remainingLength < 1) {
    return { value: null, remainingLength };
  }
  if (typeof value === "string") {
    const cutString = value.substring(0, remainingLength);
    if (remainingLength - cutString.length === 0) {
      return {
        value: (
          <>
            {cutString}...&nbsp;
            <Button
              type={"button"}
              unstyled={true}
              onClick={() => setHidden(false)}
            >
              View more
            </Button>
          </>
        ),
        remainingLength: 0,
      };
    }
    return {
      value: cutString,
      remainingLength: remainingLength - cutString.length,
    };
  } else if (Array.isArray(value)) {
    let newValArr = [];
    for (let i = 0; i < value.length; i++) {
      let splitVal = trimField(value[i], remainingLength, setHidden);
      remainingLength = splitVal.remainingLength;
      newValArr.push(
        <React.Fragment key={`arr-${i}-${splitVal.value}`}>
          {splitVal.value}
        </React.Fragment>,
      );
    }
    return { value: newValArr, remainingLength: remainingLength };
  } else if (React.isValidElement(value) && value.props.children) {
    let childrenCopy: ReactNode;
    if (Array.isArray(value.props.children)) {
      childrenCopy = [...value.props.children];
    } else {
      childrenCopy = value.props.children;
    }
    let split = trimField(childrenCopy, remainingLength, setHidden);
    const newElement = React.cloneElement(
      value,
      { ...value.props },
      split.value,
    );
    return { value: newElement, remainingLength: split.remainingLength };
  }
  return { value, remainingLength: remainingLength };
};
