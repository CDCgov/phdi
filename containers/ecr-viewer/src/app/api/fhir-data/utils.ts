import * as fs from 'fs';
import * as path from 'path';
import yaml from 'js-yaml';

export function loadYamlConfig() {
  const filePath = path.join(process.cwd(), 'src/app/api/fhir-data/fhirPath.yml');
  const fileContents = fs.readFileSync(filePath, 'utf8');
  const data = yaml.load(fileContents);
  return data;
}
