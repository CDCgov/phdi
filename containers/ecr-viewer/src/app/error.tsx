"use client"; // Error components must be Client Components

import { useEffect } from "react";

/**
 * Renders an Error component to display when an error occurs.
 * @param props - The props object.
 * @param props.error - The error object, possibly with a digest.
 * @returns - Returns the JSX element for the Error component.
 */
export default function Error({
  error,
}: {
  error: Error & { digest?: string };
}) {
  useEffect(() => {
    // Log the error to an error reporting service
    console.error(error);
  }, [error]);

  return (
    <div>
      <h2>Something went wrong!</h2>
    </div>
  );
}
