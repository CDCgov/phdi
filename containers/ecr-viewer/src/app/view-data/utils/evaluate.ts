import {
  evaluate as fhirPathEvaluate,
  type Model,
  type Path,
  type UserInvocationTable,
} from "fhirpath";
import { Context } from "node:vm";

let cache: { [key: string]: any } = {};

/**
 * Evaluates a FHIRPath expression on the provided FHIR data.
 * @param fhirData - The FHIR data to evaluate the FHIRPath expression on.
 * @param path - The FHIRPath expression to evaluate.
 * @param [context] - Optional context object to provide additional data for evaluation.
 * @param [model] - Optional model object to provide additional data for evaluation.
 * @param [options] - Optional options object for additional configuration.
 * @param [options.resolveInternalTypes] - Whether to resolve internal types in the evaluation.
 * @param [options.traceFn] - Optional trace function for logging evaluation traces.
 * @param [options.userInvocationTable] - Optional table for tracking user invocations.
 * @returns - An array containing the result of the evaluation.
 */
export const evaluate = (
  fhirData: any,
  path: string | Path,
  context?: Context,
  model?: Model,
  options?: {
    resolveInternalTypes?: boolean;
    traceFn?: (value: any, label: string) => void;
    userInvocationTable?: UserInvocationTable;
  },
): any[] => {
  let key =
    JSON.stringify(fhirData) +
    ",,,,," +
    JSON.stringify(context) +
    ",,,,," +
    JSON.stringify(path);
  if (cache.hasOwnProperty(key)) {
    return cache[key];
  }
  cache[key] = fhirPathEvaluate(fhirData, path, context, model, options);
  return cache[key];
};
