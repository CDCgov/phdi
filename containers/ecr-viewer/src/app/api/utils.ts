import * as fs from "fs";
import * as path from "path";
import yaml from "js-yaml";
import { PathMappings } from "@/app/view-data/utils/utils";

/**
 * Loads the YAML configuration for path mappings from a predefined file location.
 * @returns An object representing the path mappings defined in the YAML configuration file.
 */
export function loadYamlConfig(): PathMappings {
  const filePath = path.join(process.cwd(), "src/app/api/fhirPath.yml");
  const fileContents = fs.readFileSync(filePath, "utf8");
  return <PathMappings>yaml.load(fileContents);
}

/**
 * Converts stream data to json data
 * @param stream - The input stream that provides JSON data in chunks. The stream
 *   should implement the async iterable protocol to allow for-await-of
 *   iteration over its data chunks.
 * @returns A promise that resolves to the JSON-parsed object from the accumulated
 *  stream data. The specific structure of this object depends on the JSON
 *  content of the stream.
 */
export async function streamToJson(stream: any) {
  let rawData = "";
  for await (const chunk of stream) {
    rawData += chunk;
  }
  return JSON.parse(rawData);
}
