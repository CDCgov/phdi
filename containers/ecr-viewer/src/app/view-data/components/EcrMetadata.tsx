import { DisplayData, ReportableConditions } from "../../utils";
import { Fragment } from "react";

import {
  AccordianSection,
  AccordianH4,
  AccordianDiv,
} from "../component-utils";
import { SectionConfig } from "./SideNav";
import { Table } from "@trussworks/react-uswds";

interface EcrMetadataProps {
  rrDetails: ReportableConditions;
  eicrDetails: DisplayData[];
  eCRSenderDetails: DisplayData[];
}

export const ecrMetadataConfig: SectionConfig = new SectionConfig(
  "eCR Metadata",
  ["RR Details", "eICR Details", "eCR Sender Details"],
);

interface ReportableConditionsList {
  [condition: string]: {
    [trigger: string]: Set<string>; // Maps a trigger to a set of locations
  };
}

const convertDictionaryToRows = (dictionary: ReportableConditionsList) => {
  if (!dictionary) return [];
  const rows: JSX.Element[] = [];
  Object.entries(dictionary).forEach(([condition, triggers], _) => {
    Object.entries(triggers).forEach(([trigger, locations], triggerIndex) => {
      const locationsArray = Array.from(locations);
      locationsArray.forEach((location, locationIndex) => {
        const isConditionRow = triggerIndex === 0 && locationIndex === 0;
        const isTriggerRow = locationIndex === 0;
        const conditionCell = isConditionRow ? (
          <td
            rowSpan={Object.keys(triggers).reduce(
              (acc, key) => acc + Array.from(triggers[key]).length,
              0,
            )}
          >
            {condition}
          </td>
        ) : null;
        const triggerCell = isTriggerRow ? (
          <td rowSpan={locationsArray.length}>{trigger}</td>
        ) : null;

        rows.push(
          <tr key={`${condition}-${trigger}-${location}`}>
            {conditionCell}
            {triggerCell}
            <td>{location}</td>
          </tr>,
        );
      });
    });
  });

  return rows;
};

/**
 * Functional component for displaying eCR metadata.
 * @param props - Props containing eCR metadata.
 * @param props.rrDetails - The reportable conditions details.
 * @param props.eicrDetails - The eICR details.
 * @param props.eCRSenderDetails - The eCR sender details.
 * @returns The JSX element representing the eCR metadata.
 */
const EcrMetadata = ({
  rrDetails,
  eicrDetails,
  eCRSenderDetails,
}: EcrMetadataProps) => {
  return (
    <AccordianSection>
      <AccordianH4>
        <span id={ecrMetadataConfig.subNavItems?.[0].id}>
          {ecrMetadataConfig.subNavItems?.[0].title}
        </span>
      </AccordianH4>
      <AccordianDiv>
        <Table bordered caption="Reportibility Summary" className="rrTable">
          <thead>
            <tr>
              <th className="reportability_summary_header">
                Reportable Condition
              </th>
              <th>RCKMS Rule Summary</th>
              <th className="reportability_summary_header">
                Jurisdiction Sent eCR
              </th>
            </tr>
          </thead>
          <tbody>{convertDictionaryToRows(rrDetails)}</tbody>
        </Table>
        <div className={"padding-bottom-1"} />
        <AccordianH4>
          <span id={ecrMetadataConfig.subNavItems?.[1].id}>
            {ecrMetadataConfig.subNavItems?.[1].title}
          </span>
        </AccordianH4>
        {eicrDetails.map(({ title, value }) => {
          return (
            <Fragment key={title}>
              <div className="grid-row">
                <div className="data-title">{title}</div>
                <div className="grid-col-auto text-pre-line">{value}</div>
              </div>
              <div className={"section__line_gray"} />
            </Fragment>
          );
        })}
        <div className={"padding-bottom-1"} />
        <AccordianH4>
          <span id={ecrMetadataConfig.subNavItems?.[2].id}>
            {ecrMetadataConfig.subNavItems?.[2].title}
          </span>
        </AccordianH4>
        {eCRSenderDetails.map(({ title, value }) => {
          return (
            <Fragment key={title}>
              <div className="grid-row">
                <div className="data-title">{title}</div>
                <div className="grid-col-auto text-pre-line">{value}</div>
              </div>
              <div className={"section__line_gray"} />
            </Fragment>
          );
        })}
      </AccordianDiv>
    </AccordianSection>
  );
};

export default EcrMetadata;
