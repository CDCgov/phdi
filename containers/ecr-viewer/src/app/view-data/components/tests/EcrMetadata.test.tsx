import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import EcrMetadata from "../EcrMetadata";
import { DisplayData } from "@/app/utils";

describe("ECR Metadata", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const rrDetails: DisplayData[] = [
      {
        title: "Reportable Condition(s)",
        value: "Disease caused by severe acute respiratory syndrome coronavirus 2 (disorder)"
      },      {
        title: "RCKMS Trigger Summary",
        value: "COVID-19 (as a diagnosis or active problem)"
      },      {
        title: "Jurisdiction(s) Sent eCR",
        value: "California Department of Public Health"
      },
    ];

    const eicrDetails: DisplayData[] = [
      {
        title: "eICR Identifier",
        value: "1dd10047-2207-4eac-a993-0f706c88be5d"
      },
    ];
    const ecrSenderDetails: DisplayData[] = [
      {
        title: "Date/Time eCR Created",
        value: "2022-05-14T12:56:38Z"
      },    {
        title: "Sender Software",
        value: ""
      },    {
        title: "Sender Facility Name",
        value: "PRM- Palmdale Regional Medical Center"
      },    {
        title: "Facility Address",
        value: "38600 Medical Center Drive\nPalmdale, CA\n93551, USA"
      },    {
        title: "Facility Contact",
        value: "(661)382-5000"
      },    {
        title: "Facility ID",
        value: "2.16.840.1.113883.4.6"
      },
    ];

    container = render(
      <EcrMetadata
        eicrDetails={eicrDetails}
        eCRSenderDetails={ecrSenderDetails}
        rrDetails={rrDetails}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
