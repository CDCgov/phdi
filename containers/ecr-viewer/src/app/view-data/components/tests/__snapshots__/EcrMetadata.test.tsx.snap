// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ECR Metadata should match snapshot 1`] = `
<div>
  <div>
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <h4
          class="usa-summary-box__heading padding-y-105"
          id="summary-box-key-information"
        >
          <span
            id="rr-details"
          >
            RR Details
          </span>
        </h4>
        <div
          class="usa-summary-box__text"
        >
          <table
            class="usa-table rrTable"
            data-testid="table"
          >
            <caption>
              Reportibility Summary
            </caption>
            <thead>
              <tr>
                <th>
                  Reportable Condition
                </th>
                <th>
                  RCKMS Rule Summary
                </th>
                <th>
                  Jurisdiction Sent eCR
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  rowspan="99"
                >
                  0
                </td>
                <td
                  rowspan="23"
                >
                  title
                </td>
                <td>
                  R
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  p
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  b
                </td>
              </tr>
              <tr>
                <td>
                  l
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  C
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  (
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  )
                </td>
              </tr>
              <tr>
                <td
                  rowspan="76"
                >
                  value
                </td>
                <td>
                  D
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  b
                </td>
              </tr>
              <tr>
                <td>
                  y
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  v
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  p
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  y
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  y
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  m
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  v
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  2
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  (
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  )
                </td>
              </tr>
              <tr>
                <td
                  rowspan="64"
                >
                  1
                </td>
                <td
                  rowspan="21"
                >
                  title
                </td>
                <td>
                  R
                </td>
              </tr>
              <tr>
                <td>
                  C
                </td>
              </tr>
              <tr>
                <td>
                  K
                </td>
              </tr>
              <tr>
                <td>
                  M
                </td>
              </tr>
              <tr>
                <td>
                  S
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  T
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  g
                </td>
              </tr>
              <tr>
                <td>
                  g
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  S
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  m
                </td>
              </tr>
              <tr>
                <td>
                  m
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  y
                </td>
              </tr>
              <tr>
                <td
                  rowspan="43"
                >
                  value
                </td>
                <td>
                  C
                </td>
              </tr>
              <tr>
                <td>
                  O
                </td>
              </tr>
              <tr>
                <td>
                  V
                </td>
              </tr>
              <tr>
                <td>
                  I
                </td>
              </tr>
              <tr>
                <td>
                  D
                </td>
              </tr>
              <tr>
                <td>
                  -
                </td>
              </tr>
              <tr>
                <td>
                  1
                </td>
              </tr>
              <tr>
                <td>
                  9
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  (
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  g
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  v
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  p
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  b
                </td>
              </tr>
              <tr>
                <td>
                  l
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  m
                </td>
              </tr>
              <tr>
                <td>
                  )
                </td>
              </tr>
              <tr>
                <td
                  rowspan="62"
                >
                  2
                </td>
                <td
                  rowspan="24"
                >
                  title
                </td>
                <td>
                  J
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  d
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  (
                </td>
              </tr>
              <tr>
                <td>
                  s
                </td>
              </tr>
              <tr>
                <td>
                  )
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  S
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  C
                </td>
              </tr>
              <tr>
                <td>
                  R
                </td>
              </tr>
              <tr>
                <td
                  rowspan="38"
                >
                  value
                </td>
                <td>
                  C
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  l
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  f
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  D
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  p
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  r
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  m
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  n
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  o
                </td>
              </tr>
              <tr>
                <td>
                  f
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  P
                </td>
              </tr>
              <tr>
                <td>
                  u
                </td>
              </tr>
              <tr>
                <td>
                  b
                </td>
              </tr>
              <tr>
                <td>
                  l
                </td>
              </tr>
              <tr>
                <td>
                  i
                </td>
              </tr>
              <tr>
                <td>
                  c
                </td>
              </tr>
              <tr>
                <td>
                   
                </td>
              </tr>
              <tr>
                <td>
                  H
                </td>
              </tr>
              <tr>
                <td>
                  e
                </td>
              </tr>
              <tr>
                <td>
                  a
                </td>
              </tr>
              <tr>
                <td>
                  l
                </td>
              </tr>
              <tr>
                <td>
                  t
                </td>
              </tr>
              <tr>
                <td>
                  h
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="padding-bottom-1"
          />
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="summary-box-key-information"
          >
            <span
              id="eicr-details"
            >
              eICR Details
            </span>
          </h4>
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              eICR Identifier
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              1dd10047-2207-4eac-a993-0f706c88be5d
            </div>
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="padding-bottom-1"
          />
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="summary-box-key-information"
          >
            <span
              id="ecr-sender-details"
            >
              eCR Sender Details
            </span>
          </h4>
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Date/Time eCR Created
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              2022-05-14T12:56:38Z
            </div>
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Sender Software
            </div>
            <div
              class="grid-col-auto text-pre-line"
            />
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Sender Facility Name
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              PRM- Palmdale Regional Medical Center
            </div>
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Facility Address
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              38600 Medical Center Drive
Palmdale, CA
93551, USA
            </div>
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Facility Contact
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              (661)382-5000
            </div>
          </div>
          <div
            class="section__line_gray"
          />
          <div
            class="grid-row"
          >
            <div
              class="data-title"
            >
              Facility ID
            </div>
            <div
              class="grid-col-auto text-pre-line"
            >
              2.16.840.1.113883.4.6
            </div>
          </div>
          <div
            class="section__line_gray"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
